### PR TITLE
[REEF-962] Use enum for language

### PR DESCRIPTION
**Pull request by jwang98052**
Added enum definition for Languages
Set proper default where the language is used
Updated test cases

JIRA: [REEF-962](https://issues.apache.org/jira/browse/REEF-962)

This closes #

### CODE REVIEW

File: lang/cs/Org.Apache.REEF.Tang/Types/Languages.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+namespace Org.Apache.REEF.Tang.Types
+{
+    public enum Languages
```
Comment (afchung) [inline lang/cs/Org.Apache.REEF.Tang/Types/Languages.cs]: Please add documentation.

Comment (afchung) [inline lang/cs/Org.Apache.REEF.Tang/Types/Languages.cs]: Rename to ``Language``.

File: lang/cs/Org.Apache.REEF.Tang/Implementations/ClassHierarchy/NamedParameterNodeImpl.cs [hunk 2]
```diff
@@ -54,7 +54,7 @@ public class NamedParameterNodeImpl : AbstractNode, INamedParameterNode
         public NamedParameterNodeImpl(INode parent, String simpleName,
             String fullName, String fullArgName, String simpleArgName, bool isSet, bool isList,
             String documentation, String shortName, String[] defaultInstanceAsStrings)
-            : this(parent, simpleName, fullName, simpleArgName, simpleArgName, isSet, isList, documentation, shortName, defaultInstanceAsStrings, null, null)
+            : this(parent, simpleName, fullName, simpleArgName, simpleArgName, isSet, isList, documentation, shortName, defaultInstanceAsStrings, null, Languages.Cs)
```
Comment (afchung) [inline lang/cs/Org.Apache.REEF.Tang/Implementations/ClassHierarchy/NamedParameterNodeImpl.cs]: Why not use default parameter?

File: lang/cs/Org.Apache.REEF.Tang/Org.Apache.REEF.Tang.csproj [hunk 3]
```diff
@@ -128,6 +128,7 @@ under the License.
     <Compile Include="Types\INode.cs" />
     <Compile Include="Types\IPackageNode.cs" />
     <Compile Include="Types\ITraversable.cs" />
+    <Compile Include="Types\Languages.cs" />
```
Comment (afchung) [inline lang/cs/Org.Apache.REEF.Tang/Org.Apache.REEF.Tang.csproj]: I believe this fits more in ``Util`` than in ``Types``.

File: lang/cs/Org.Apache.REEF.Tang/Protobuf/ProtocolBufferClassHierarchy.cs [hunk 4]
```diff
@@ -307,11 +304,11 @@ private void AddAlias(INamedParameterNode np)
             if (np.GetAlias() != null && !np.GetAlias().Equals(""))
```
Comment (afchung) [inline lang/cs/Org.Apache.REEF.Tang/Protobuf/ProtocolBufferClassHierarchy.cs]: string.Empty.

File: lang/cs/Org.Apache.REEF.Tang/Protobuf/ProtocolBufferClassHierarchy.cs [hunk 5]
```diff
@@ -338,10 +335,21 @@ private static void ParseSubHierarchy(INode parent, Org.Apache.REEF.Tang.Protobu
 
                 if (np.alias_name != null && np.alias_language != null)
```
Comment (afchung) [inline lang/cs/Org.Apache.REEF.Tang/Protobuf/ProtocolBufferClassHierarchy.cs]: ``string,IsNullOrWhiteSpace``

**Comment by afchung** [comment]
@jwang98052 Looks good in general. Only minor comments. Thanks!

File: lang/cs/Org.Apache.REEF.Tang/Types/Languages.cs (hunk 1 above)
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Tang/Types/Languages.cs]: done

File: lang/cs/Org.Apache.REEF.Tang/Implementations/ClassHierarchy/NamedParameterNodeImpl.cs (hunk 2 above)
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Tang/Implementations/ClassHierarchy/NamedParameterNodeImpl.cs]: Done

File: lang/cs/Org.Apache.REEF.Tang/Org.Apache.REEF.Tang.csproj (hunk 3 above)
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Tang/Org.Apache.REEF.Tang.csproj]: moved

File: lang/cs/Org.Apache.REEF.Tang/Protobuf/ProtocolBufferClassHierarchy.cs (hunk 4 above)
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Tang/Protobuf/ProtocolBufferClassHierarchy.cs]: done

File: lang/cs/Org.Apache.REEF.Tang/Protobuf/ProtocolBufferClassHierarchy.cs (hunk 5 above)
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Tang/Protobuf/ProtocolBufferClassHierarchy.cs]: done

**Comment by jwang98052** [comment]
@afchung I have addressed review comments and pushed in the change.

**Comment by afchung** [comment]
Looks good, I'll test and merge.

**Pull request closed without merging**